src/include/txn_inline.h

```diff
@@ -1061,8 +1061,12 @@ __wt_txn_begin(WT_SESSION_IMPL *session, const char *cfg[])
 
     WT_RET(__wt_txn_config(session, cfg));
 
-    /* Allocate a snapshot if required. */
-    if (txn->isolation == WT_ISO_SNAPSHOT) {
+    /*
+     * Allocate a snapshot if required or update the existing snapshot for non autocommit
+     * transaction. As there is no need of updating snapshot for autocommit transactions as they are
+     * committed at the end of the operation.
+     */
+    if (txn->isolation == WT_ISO_SNAPSHOT && (!autocommit || !F_ISSET(txn, WT_TXN_HAS_SNAPSHOT))) {
```

Comment:
It's a personal preference, but I find:
`if (txn->isolation == WT_ISO_SNAPSHOT && !(autocommit && F_ISSET(txn, WT_TXN_HAS_SNAPSHOT)))`

Easier to understand. Leave it as you had it if you prefer - doubly since you added a comment (thanks).

Reply:
Changed as you suggested.